arch/arm/slide_neon.c

```diff
@@ -18,15 +18,15 @@
 #include "../../deflate.h"
 
 /* SIMD version of hash_chain rebase */
-static inline void slide_hash_chain(Pos *table, unsigned int entries, uint16_t window_size) {
+static inline void slide_hash_neon_chain(Pos *table, uint32_t entries, uint16_t wsize) {
```

Comment:
The function name is just plain awkward... "hash chain" is one word, you can't split it as two words without changing meaning.

Reply:
Power was using `slide_hash_power8_loop`. I can change it to use "loop" instead of "chain" if you prefer.

Reply:
I have no opinion about better name, except that it should reflect what the function actually does...

Reply:
Come on man make a suggestion.

Reply:
"loop" is quite self-explanatory... But it's not up to me to decide alone what is the very best name of the function...

Reply:
I am just asking for a suggestion..

Reply:
I know... It's early morning here and for last few weeks I have been working on project written in another programming language, so my head is pretty empty right now... As I said, I will wait for opinion from other contributors...

Reply:
I also think the current naming is a bit unexpected. What if we simply use `slide_hash_chain` for this on all archs? The actual function that will be called elsewhere will just be `slide_hash_{power8,neon,avx2,..}`, which will have the proper arch extension in the name.

Other option is to simply swap the last two strings and call this `slide_hash_chain_neon`, or the equivalent for the other arch implementations. It reads more naturally.

Reply:
I have renamed it to `slide_hash_chain` as you suggested.